pages/settings/audits.tsx

```diff
@@ -1,30 +1,58 @@
 import { type GetServerSidePropsContext } from "next";
+import { Fragment, useState } from "react";
 import SettingsLayout from "@/layouts/Settings";
 import { getServerSideSession } from "@/utils/session";
+import { trpc } from "@/utils/trpc";
 import { User } from "@prisma/client";
+import { useQuery } from "@tanstack/react-query";
 import AuditLogs from "@/components/projects/AuditLogs";
 import { Button } from "@/components/theme";
 import Audit from "@/lib/audit";
 
 interface AuditSettingsProps {
   user: User;
-  logs: any;
 }
 
-const AuditSettings = ({ user, logs }: AuditSettingsProps) => {
+const AuditSettings = ({ user }: AuditSettingsProps) => {
+  const [page, setPage] = useState(0);
+  const {
+    data: auditLogs,
+    fetchNextPage,
+    isLoading,
+    status,
+    isFetchingNextPage,
+    hasNextPage
+  } = trpc.audits.paginate.useInfiniteQuery(
+    { limit: 20 },
+    {
+      getNextPageParam: (lastPage) => lastPage.nextCursor,
+    },
+  );
+
   return (
     <SettingsLayout tab={"audit"} user={user}>
       <h3 className="mb-8 text-lg ">Audit logs</h3>
 
       <div className="w-full lg:w-3/5">
-        <AuditLogs logs={logs} user={user} />
+        {isLoading ? (
+          "loading..."
+        ) : (
+            auditLogs?.pages.map((page, index) => (
+              <Fragment key={index}>
+                <AuditLogs logs={page.logs} user={user} />
+              </Fragment>
```

Comment:
Looks like re-rendering breaks the UI, instead we can use state and `useEffect` hook on `AuditLog` component?
![Screen Shot 2023-02-01 at 6 07 29 PM](https://user-images.githubusercontent.com/3684236/216198061-7ee6bdcc-0bda-4d24-b936-403af8db989d.png)

Reply:
Gonna look on that.